### ISSUE TITLE
NCTL: staging missed upgrade and restarting with post-upgrade hash

### ISSUE THREAD

**Issue by TomVasile** [casper-network/casper-node#2087]
### Issue:
Seeing nodes failing to restart after missing an upgrade, staging the upgrade, and attempting to restart. This happens deterministically when using a `trusted_hash` from post upgrade. 
**NOTE: This does not occur when using a `trusted_hash` prior to the upgrade** 

I dumped the logs on two separate occasions and attached the associated dumps here.

Code is not merged yet but can be found here: https://github.com/TomVasile/casper-node/tree/sre-issue-112

### Summary:

- Nodes 2thru9 were upgraded, while nodes 1 and 10 were left behind
- Nodes 1 and 10 were then staged and rebooted with a trusted hash post-upgrade.
- Nodes 1&10 will exit. 
    - **NOTE: sometimes immediately, sometimes after a bit**

### Steps to Reproduce:
Enable Post-Upgrade Hash:
```
vi $NCTL/sh/scenarios-upgrades/upgrade_scenario_04.sh

# Comment out these lines in _main()
PRE_UPGRADE_HASH="$($(get_path_to_client) get-block --node-address "$(get_node_address_rpc '2')" | jq -r '.result.block.hash')"

_step_09 "$STAGE_ID" "$ACTIVATION_POINT" "$PRE_UPGRADE_HASH"

# Uncomment these lines in _main()
POST_UPGRADE_HASH="$($(get_path_to_client) get-block --node-address "$(get_node_address_rpc '2')" | jq -r '.result.block.hash')"

_step_09 "$STAGE_ID" "$ACTIVATION_POINT" "$POST_UPGRADE_HASH"
```

**NOTE: Reverse the above to use a pre-upgrade hash which will let the nodes restart for comparison. The test may still fail due to another issue captured by ZenHub issue: [syncing issue after upgrade](https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/governance/69)**

```
nctl-compile

nctl-stage-set-remotes 1.3.3

mkdir -p "$(get_path_to_stage 1)"

cat <<EOF > "$(get_path_to_stage_settings 1)"
export NCTL_STAGE_SHORT_NAME="YOUR-SHORT-NAME"

export NCTL_STAGE_DESCRIPTION="YOUR-DESCRIPTION"

export NCTL_STAGE_TARGETS=(
    "1_3_3:remote"
    "1_4_3:local"
)
EOF

nctl-stage-build-from-settings

bash -i $NCTL/sh/scenarios-upgrades/upgrade_scenario_04.sh
```

**Comment by TomVasile** [comment]
[dump.tar.gz](https://app.zenhub.com/files/263892105/6a96e192-1783-44d5-b6a9-5e3a8eb98a08/download)
[dump2.tar.gz](https://app.zenhub.com/files/263892105/3dc2b999-2fb1-47f8-9996-f01e39686866/download)

**Comment by Fraser999** [comment]
Fixed via #2101.
